spring-pulsar/src/main/java/org/springframework/pulsar/PulsarException.java

```diff
@@ -33,4 +34,8 @@ public PulsarException(String msg, Throwable cause) {
 		super(msg, cause);
 	}
 
+	public PulsarException(Exception exception) {
```

Comment:
I think we could open this up to `Throwable` which will also keep it consistent w/ the parent and the other `<init>s`. Wdyt?

Reply:
Doesn't opening `PulsarException` to `Throwable` implies opening it up for those errors that are unrecoverable like an `OutOfMemoryError` ? Which is not the point of `PulsarException` it is for handling Pulsar Exceptions.
I would personally stick with exception as most if not all Pulsar related exception we want to catch will be subclasses of `Exception`.

Reply:
We would not be catching throwable but rather just using it as the type. I was mostly leaning on consistency w/ the already existing `<init>` which do take in `Throwable` as well as the parent `NestedRuntimeException` that also codes to `Throwable`.

Reply:
Not catching obviously but the `PulsarException` constructor would imply catching `Throwable` in a way.